internal/cmd/rootcmd/env.go

```diff
@@ -13,6 +13,18 @@ const envKeyPrefix = "CONTROLLER_"
 
 // bindEnvVars, for each flag defined on `cmd` (local or parent persistent), looks up the corresponding environment
 // variable and (if the flag is unset) takes that environment variable value as the flag value.
+//
+// Why this is custom-written and not imported from spf13/viper:
+// KIC 1.x used `spf13/viper` in order to provide equivalent (in fact, 100% compatible)
+// behavior. In KIC 2.x we decided to drop spf13/viper in favor of a hand-written mechanism for the following reasons:
+// - viper-cobra integration, per [1] and [2], requires a pflag.FlagSet.VisitAll call that binds viper values to flags
+//   one by one,
+// - viper is a comparably heavy dependency - it pulls in several additional features that KIC does not need (e.g.
+//   configfile support),
+// - the viper-cobra integration described in the 1st bullet point required more code than the whole mechanism below.
+// 
```

Comment:
From the complaints of linter we see there is an extra space after `//`.

Reply:
This has already been addressed, thanks!